### DISCUSSION TITLE
Pedestrian Collision Prediction

### DISCUSSION THREAD

**Discussion by smohiudd** [kraina-ai/srai#422] [General]
I did some work using `srai` embeddings to train a linear classifier to detect pedestrian collision locations: https://github.com/smohiudd/pedestrian-collision-prediction

Would love some feedback on my work!

**Comment by RaczeQ** [comment]
Hello Saadiq!

I'm delighted to see your project and the fact that you've been able to apply the spatial embeddings in a real-data example.

I've read the publication and the code inside the repository. I'm impressed with the visualizations and thorough descriptions of the models with the provided sources.

However, the repository's lack of a `requirements.txt` file gave me a little trouble before running the notebooks.
I've never used the `svd` or `lonboard` library so I can't comment on the use of those 😅 
From what I've seen, you have successfully changed our example for the GeoVex model to generate your embeddings. I hope it was easy to follow and understand 😄 
In the future the `srai` library will have a faster `osm.pbf` reader based on `DuckDB` with geometry filtering so it won't require any manual step with osmium extracts clipping 😉

For final predictions and visualizations, I'd select only hexagons with roads intersecting them since that's where accidents occur.

If I wanted to nitpick, then I would change the sampling logic for negative examples - the current logic with sampling from the roads graph is clever (and that was my initial idea as well), but seeing how many accidents from the dataset occur at the intersections, I'd force the sampler to include more intersections in the negative dataset. Maybe even doing some stratification based on road type, to exclude smaller roads near homes.
I'm also not sure if embedding vector size of 5 is big enough to distinguishably compress the data, but that would require more tests of course.

In summary, I think it's a nicely prepared small data science project to start interacting with geospatial embeddings. Thank you for this contribution, Saadiq!

**Reply by smohiudd** [reply]
Hello Kamil,

Thank you very much for you feedback and critique (very much appreciate the nit picking)! 

Good reminder about the requirements file - something I overlooked and will make sure I include. 

Even though I know `srai` is in active development and evolving, the documentation was great and was easy to follow and use. For sure one of the easier parts of this experiment.

The negative sampling logic is something I've been struggling with and still not sure the best approach. As you know I went with a 1:1 ratio but that's probably not ideal for a rare occurrence event like a collision. I'm still playing with what do about this. But I like the stratification idea and will try that. 

I tried different embedding sizes between 5 and 30 and didn't notice much improvement in the model so went with the most compact embedding that got decent results. 

Thanks again for your great work on this library. Will be following this development closely and look forward to more releases!
